tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -330,16 +330,30 @@ public async Task ConfidentialClientUsingAdfsAsync()
             MockHttpManager httpManager,
             X509Certificate2 cert,
             int tokenResponses,
-            TelemetryCallback telemetryCallback = null)
+            TelemetryCallback telemetryCallback = null,
+            ClientAssertion assertion = null)
         {
-            var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
-                                                          .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
-                                                          .WithRedirectUri(MsalTestConstants.RedirectUri)
-                                                          .WithCertificate(cert)
-                                                          .WithHttpManager(httpManager)
-                                                          .WithTelemetry(telemetryCallback)
-                                                          .BuildConcrete();
-
+            ConfidentialClientApplication app;
+            if (assertion != null)
+            {
+                app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                              .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
```

Comment:
DRY (don't repeat yourself) - you can create a builder without client assertion and without certificate. Then use .WithClientAssertion or .WIthCertificate